#!/usr/bin/env python3
PACKAGE = "asl_turtlebot"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("k1",       double_t,    0, "Pose Controller k1", 1.0,  0., 2.0) # Original 0.8
gen.add("k2",       double_t,    0, "Pose Controller k2", 1.0,  0., 2.0) # Original 0.4
gen.add("k3",       double_t,    0, "Pose Controller k3", 1.0,  0., 2.0) # Original 0.4

gen.add("distance_stop_sign",       double_t,    0, "Threshold Distance stop_sign", .9,  0., 2.0)
gen.add("distance_fire_hydrant",       double_t,    0, "Threshold Distance fire_hydrant", .9,  0., 2.0)
gen.add("distance_person",       double_t,    0, "Threshold Distance person", .9,  0., 2.0)
gen.add("distance_cow",       double_t,    0, "Threshold Distance cow", .9,  0., 2.0)

gen.add("confidence_stop_sign",       double_t,    0, "Threshold Confidence stop_sign", .1,  0., 1.0)
gen.add("confidence_fire_hydrant",       double_t,    0, "Threshold Confidence fire_hydrant", .1,  0., 1.0)
gen.add("confidence_person",       double_t,    0, "Threshold Confidence person", .1,  0., 1.0)
gen.add("confidence_cow",       double_t,    0, "Threshold Confidence cow", .1,  0., 1.0)


exit(gen.generate(PACKAGE, "navigator", "Navigator"))

